whylog/assistant/regex_assistant/regex.py

```diff
@@ -3,13 +3,75 @@
 """
 
 import re
+from collections import deque
+
+import six
 
 from whylog.assistant.regex_assistant.exceptions import NotMatchingRegexError
+from whylog.assistant.span import Span
+from whylog.assistant.span_list import SpanList
 
 special_characters = frozenset('.^$*+{}?[]|()]')
 group_pattern = re.compile(r"[a-zA-Z]+|[0-9]+|\s+|[\W]+")
 
 
+def group_spans_from_regex(regex, text):
+    """
+    Creates spans from regex groups in regex corresponding to text.
+
+    Assumption: regex matches text.
+    """
+
+    if len(regex) < 2:
```

Comment:
if regex has one character, it has no groups. I can change it to `if len(regex) < 1`, because I need this condition in line 38. but, for 1-length regex there is also no groups, so I would leave it.